File: C/05 - primeOrComposite.c
#include <stdio.h>
#include <math.h>

int main() {
    int n, i = 2;
    printf("Enter any number: ");
    scanf("%d", &n);
    int r = (n + 2) / 2;
    while (1) {
        if (i > r) { printf("%d is prime", n); break; }
        if (n % i == 0) { printf("%d is composite", n); break; }
        i++;
    }
    return 0;
}

File: C#/05 - primeOrComposite.cs
using System;

public class primeOrComposite
{
    public static void Main(string[] args)
    {
        int n, i = 2;
        Console.Write("Enter any number: ");
        n = Convert.ToInt32(Console.ReadLine());
        int r = (int)Math.Sqrt(n);
        while (true) {
            if (i > r) { Console.WriteLine(n + " is prime"); break; }
            if (n % i == 0) { Console.WriteLine(n + " is composite"); break; }
            i++;
        }
    }
}